### PR TITLE
Fixes for Peckshield audit

### PR DESCRIPTION
**Pull request by stevenwal**
This is to deliver fixes for Peckshield audit for Ribbon Treasury contracts

### CODE REVIEW

**Comment by stevenwal** [comment]
PVE-004
In initialization, the contract sets the variable `vaultState.lastLockedAmount` which is different from the one we used for `totalBalance()` which is `vaultState.lockedAmount`. Hence, there is no double counting here.

File: test/RibbonTreasuryVault.ts [hunk 1]
```diff
@@ -87,140 +87,145 @@ describe("RibbonTreasuryVault", () => {
...
-    mintConfig: {
-      contractOwnerAddress: SUSHI_OWNER_ADDRESS[chainId],
-    },
-    period: 180,
-    multiplier: 150,
-    premiumDecimals: 6,
-    maxDepositors: 30,
-    availableChains: [CHAINID.ETH_MAINNET],
-  });
+  // behavesLikeRibbonOptionsVault({
```
Comment (kenchangh) [inline test/RibbonTreasuryVault.ts:94]: can just remove this entirely

File: contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol [hunk 2]
```diff
@@ -545,6 +567,24 @@ contract RibbonTreasuryVault is
...
+                numShares,
+                currentRound != 1
+                    ? roundPricePerShare[currentRound - 1]
+                    : 10**vaultDecimals,
+                vaultDecimals
+            );
+
+        if (userBalance > withdrawAmount) {
+            uint256 totalDeposit = userBalance.sub(withdrawAmount);
+            require(totalDeposit >= minDeposit, "Minimum deposit not reached");
```
Comment (kenchangh) [inline contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol:585]: Minimum withdrawal not reached

File: contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol [hunk 3]
```diff
@@ -496,6 +515,8 @@ contract RibbonTreasuryVault is
             depositAmount = newAmount;
         }
 
+        // depositAmount > minDeposit
```
Comment (kenchangh) [inline contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol:518]: can remove this

**Pull request closed without merging**